aea/helpers/base.py

```diff
@@ -563,7 +563,7 @@ def find_topological_order(adjacency_list: Dict[T, Set[T]]) -> List[T]:
     # compute the topological order
     queue: Deque[T] = deque()
     order = []
-    queue.extendleft(sorted(roots))  # type: ignore
+    queue.extendleft(sorted(roots, key=str))  # type: ignore
```

Comment:
this guarantees the ordering of root nodes is deterministic, and hence the returned `List[T]` of `find_topological_order` will be as well.